test/distributed/_tensor/test_math_ops.py

```diff
@@ -150,6 +151,44 @@ def test_full_shard_math_ops(self):
             actual_local_res = actual_rs.to_local()
             self.assertEqual(actual_local_res, expect_rs)
 
+    @with_comms
+    def test_layer_norm(self):
+        device_mesh = self.build_device_mesh()
+
+        # NLP example from pytorch docs
+        # https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html
+        batch, sentence_length, embedding_dim = 20, 5, 10
+        x = torch.rand(batch, sentence_length, embedding_dim, device=self.device_type)
+        norm_shape_idx_list = list(range(x.ndim))
+        shard_dims = [-1, 0, 1, 2]
+        test_config_list = list(itertools.product(shard_dims, norm_shape_idx_list))
+
+        # normalized shape is a torch.Size object
+        for shard_dim, norm_idx in test_config_list:
+            normalized_shape = x.shape[norm_idx:]
+            layer_norm = torch.nn.LayerNorm(normalized_shape)
+            layer_norm_local = copy.deepcopy(layer_norm).to(self.device_type)
+
+            def _replicate_fn(name, module, device_mesh):
+                for name, param in module.named_parameters():
+                    if name in ["weight", "bias"]:
+                        param_dist = torch.nn.Parameter(
+                            distribute_tensor(param, device_mesh, [Replicate()])
+                        )
+                        module.register_parameter(name, param_dist)
```

Comment:
not necessary in this PR, but can you upstream this logic to TP API (parallelize_module) so that users can just use TP API to mark layernorm as Replicated.